app/screens/settings-screen/row.tsx

```diff
@@ -0,0 +1,111 @@
+import { useState } from "react"
+import { ActivityIndicator, TouchableWithoutFeedback, View } from "react-native"
+import { makeStyles, Icon, Text, Skeleton } from "@rneui/themed"
+
+type Props = {
+  title: string
+  subtitle?: string
+  subtitleShorter?: boolean
+  leftIcon: string
+  rightIcon?: string | null | React.ReactElement
+  extraComponentBesideTitle?: React.ReactElement
+  action: (() => void | Promise<void>) | null
+  rightIconAction?: () => void | Promise<void>
+  loading?: boolean
+  spinner?: boolean
+  shorter?: boolean
+}
+
+export const SettingsRow: React.FC<Props> = ({
+  title,
+  subtitle,
+  subtitleShorter,
+  leftIcon,
+  rightIcon = "",
+  action,
+  rightIconAction = action,
+  extraComponentBesideTitle = <></>,
+  loading,
+  spinner,
+  shorter,
+}) => {
+  const [hovering, setHovering] = useState(false)
+  const styles = useStyles({ hovering, shorter })
+
+  if (loading) return <Skeleton style={styles.container} animation="pulse" />
+  if (spinner)
+    return (
+      <View style={[styles.container, styles.center]}>
+        <ActivityIndicator />
+      </View>
+    )
```

Comment:
I'm confused of the difference between `loading` and `spinner`

Reply:
loading shows skeleton of the same width
spinner shows spinner (in case user has done some action - exporting transactions, deleting totp, etc)